storage/namespace/metadata.go

```diff
@@ -33,16 +35,31 @@ type metadata struct {
 // NewMetadata creates a new namespace metadata
 func NewMetadata(id ts.ID, opts Options) Metadata {
 	copiedID := checked.NewBytes(append([]byte(nil), id.Data().Get()...), nil)
-	return metadata{
+	return &metadata{
 		id:   ts.BinaryID(copiedID),
 		opts: opts,
 	}
 }
 
-func (m metadata) ID() ts.ID {
+func (m *metadata) ID() ts.ID {
 	return m.id
 }
 
-func (m metadata) Options() Options {
+func (m *metadata) Options() Options {
 	return m.opts
 }
+
+func (m *metadata) Equal(value Metadata) bool {
+	if !m.id.Equal(value.ID()) {
+		return false
+	}
+	return m.Options().Equal(value.Options())
+}
+
+func (m *metadata) Validate() error {
+	if m.id.String() == "" {
+		return fmt.Errorf("namespace id is not set")
```

Comment:
Should this be moved to the ctor instead? Doesn't look like you can change the id or the options after construction.